### PR TITLE
[risk=low][RW-7520] CreateWorkspace TierUnavailableModal if user doesn't have access

### PR DESCRIPTION
**Pull request by jmthibault79**
Old behavior -> for users without CT access, no tier dropdown appears in Create Workspace

New behavior -> if CT exists in an env, it always appears in Create Workspace.  users without access get a modal directing them to DAR if they select CT (and the selection is forced back to RT)

Create Workspace Tier 2

Create_Workspace_Tier_2

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/workspace/tier-unavailable-modal.tsx
Comment (jmthibault79) [inline ui/src/app/pages/workspace/tier-unavailable-modal.tsx:12]: name ideas?

File: ui/src/app/pages/workspace/tier-unavailable-modal.tsx
Comment (jmthibault79) [inline ui/src/app/pages/workspace/tier-unavailable-modal.tsx:16]: This text is original, so I'll check with UX

File: ui/src/app/pages/workspace/tier-unavailable-modal.tsx
Comment (jmthibault79) [inline ui/src/app/pages/workspace/tier-unavailable-modal.tsx:18]: "please complete..." matches the no-tier-access text on the Profile page

File: ui/src/app/pages/workspace/workspace-edit.spec.tsx
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.spec.tsx:96]: remove some indirection

File: ui/src/app/pages/workspace/workspace-edit.spec.tsx
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.spec.tsx:297]: this change reflects the new behavior.  IMO this still checks the "available tiers" by a different definition

File: ui/src/app/pages/workspace/workspace-edit.tsx
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:1106]: extracted from line 1187

File: ui/src/app/pages/workspace/workspace-edit.tsx
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:1119]: but this line is new

File: ui/src/app/pages/workspace/workspace-edit.tsx
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:1155]: because I need the profile

File: ui/src/app/pages/workspace/workspace-edit.tsx
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:1196]: old: show the tiers the user **has access to**

File: ui/src/app/pages/workspace/workspace-edit.tsx
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:1214]: new: show the tiers which are **present in the environment**

File: ui/src/app/pages/workspace/workspace-edit.spec.tsx
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.spec.tsx:351]: mostly duplicates the previous test

**Comment by jmthibault79** [comment]
Waiting on UX approval

File: ui/src/app/pages/workspace/unavailable-tier-modal.tsx
Comment (dmohs) [inline ui/src/app/pages/workspace/unavailable-tier-modal.tsx:14]: Any reason why not:
```suggestion
  return <Modal onRequestClose={onCancel}>
```

Comment (dmohs) [reply ui/src/app/pages/workspace/unavailable-tier-modal.tsx:14]: Because we don't have confidence yet that we aren't using `this` in hazardous ways within the codebase.

**Review by dmohs — APPROVED** [review]

**Pull request merged**
